agent/api/task.go

```diff
@@ -847,6 +857,127 @@ func (task *Task) GetTaskENI() *ENI {
 	return task.ENI
 }
 
+// SetPullStartedAt sets the task pullstartedat timestamp and returns whether
+// this field was updated or not
+func (task *Task) SetPullStartedAt(timestamp time.Time) bool {
+	task.knownStatusTimeLock.Lock()
+	defer task.knownStatusTimeLock.Unlock()
+
+	// Only set this field if it is not set
+	if task.PullStartedAt == (time.Time{}) {
+		task.PullStartedAt = timestamp
+		return true
+	}
+	return false
+}
+
+// GetPullStartedAt returns the PullStartedAt timestamp in UTC format
+func (task *Task) GetPullStartedAt() *time.Time {
+	task.knownStatusTimeLock.RLock()
+	defer task.knownStatusTimeLock.RUnlock()
+
+	if task.PullStartedAt == (time.Time{}) {
+		return nil
+	}
+
+	utcTime := task.PullStartedAt.UTC()
+	return &utcTime
+}
+
+// SetPullStoppedAt sets the task pullstoppedat timestamp
+func (task *Task) SetPullStoppedAt(timestamp time.Time) {
+	task.knownStatusTimeLock.Lock()
+	defer task.knownStatusTimeLock.Unlock()
+
+	task.PullStoppedAt = timestamp
+}
+
+// GetPullStoppedAt returns the PullStoppedAt timestamp in UTC format
+func (task *Task) GetPullStoppedAt() *time.Time {
+	task.knownStatusTimeLock.RLock()
+	defer task.knownStatusTimeLock.RUnlock()
+
+	if task.PullStoppedAt == (time.Time{}) {
+		return nil
+	}
+
+	utcTime := task.PullStoppedAt.UTC()
+	return &utcTime
+}
+
+// SetExecutionStoppedAt sets the ExecutionStoppedAt timestamp of the task
+func (task *Task) SetExecutionStoppedAt(timestamp time.Time) {
+	task.knownStatusTimeLock.Lock()
+	defer task.knownStatusTimeLock.Unlock()
+
+	task.ExecutionStoppedAt = timestamp
+}
+
+// GetExecutionStoppedAt returns the task executionStoppedAt timestamp in UTC format
+func (task *Task) GetExecutionStoppedAt() *time.Time {
+	task.knownStatusTimeLock.RLock()
+	defer task.knownStatusTimeLock.RUnlock()
+
+	if task.ExecutionStoppedAt == (time.Time{}) {
```

Comment:
Why can we just return `task.ExecutionStoppedAt` here and let the `SubmitTaskStateChange` API handle the UTC part of it? It seems like we're adding extra computation here and increasing the lock scope and I'd prefer to do those cosmetic operations outside of the lock scope.